lahja/tools/benchmark/process.py

```diff
@@ -108,21 +120,25 @@ def launch(name: str, num_events: int) -> None:
             ConnectionConfig.from_name(REPORTER_ENDPOINT)
         )
 
-        counter = itertools.count(1)
+        events = ConsumerProcess.gen_with_timeout(
+            event_bus.stream(PerfMeasureEvent, num_events=num_events),
+            timeout=5
+        )
         stats = LocalStatistic()
-        async for event in event_bus.stream(PerfMeasureEvent):
-            stats.add(RawMeasureEntry(
-                sent_at=event.sent_at,
-                received_at=time.time()
-            ))
-
-            if next(counter) == num_events:
-                await event_bus.broadcast(
-                    TotalRecordedEvent(stats.crunch(event_bus.name)),
-                    BroadcastConfig(filter_endpoint=REPORTER_ENDPOINT)
-                )
-                event_bus.stop()
-                break
+        try:
+            async for event in events:
+                stats.add(RawMeasureEntry(
+                    sent_at=event.sent_at,
+                    received_at=time.time()
+                ))
+        except asyncio.TimeoutError:
+            print('Some messages from the consumer were dropped')
```

Comment:
Isn't a dropped event something that should fail the whole benchmark?

Reply:
So, yes and no. After spending a lot of time trying to get this benchmark to work after each change I make to lahja I'm starting to think that it's a bad idea to assume every message will be received, the sending process might start sending before the receiving process starts listening.

This change prevents `perf_benchmark.py` from hanging forever if some messages are dropped. I think that in a vacuum dropped messages might be a reason to fail the benchmark, but #39 starts the processes in the right order so that they won't be dropped, so I hope that this PR can be merged as is, given that it'll be immediately replaced with #39.

Reply:
>so I hope that this PR can be merged as is, given that it'll be immediately replaced with #39.

I'm ok to merge it as is but as noted in #39, I believe we should then remove this again so that the benchmark would fail when messages get dropped.